src/wp-includes/view-transitions.js

```diff
@@ -0,0 +1,191 @@
+// This script should eventually live elsewhere, but for now it's just in `wp-includes` for simplicity.
+if ( !! window.navigation && 'CSSViewTransitionRule' in window ) {
+	const config = { __PLACEHOLDER__: true };
+
+	const getViewTransitionEntries = ( transitionType, bodyElement, articleElement ) => {
+		const isMainSlide = transitionType === 'forwards' || transitionType === 'backwards';
+		let foundMainElement = false;
+		return [
+			...Object.entries( config.globalTransitionNames || {} ).map( ( [ selector, name ] ) => {
```

Comment:
some inline comments above each of the return elements would be nice, took a bit to understand what the code was doing exactly

Reply:
Added docs in https://github.com/WordPress/wordpress-develop/pull/8370/commits/220d1b8cf4634d7537f862920b93c9d64f942876